File: go/control/gateway.go
package control

import (
	"context"

	"github.com/core-tools/hsu-core/go/api/proto"
	"github.com/core-tools/hsu-core/go/domain"
	"github.com/core-tools/hsu-core/go/logging"
	"google.golang.org/grpc"
)

func NewGRPCClientGateway(grpcClientConnection grpc.ClientConnInterface, logger logging.Logger) domain.Contract {
	grpcClient := proto.NewCoreServiceClient(grpcClientConnection)
	return &grpcClientGateway{
		grpcClient: grpcClient,
		logger:     logger,
	}
}

type grpcClientGateway struct {
	grpcClient proto.CoreServiceClient
	logger     logging.Logger
}

func (gw *grpcClientGateway) Ping(ctx context.Context) error {
	_, err := gw.grpcClient.Ping(ctx, &proto.PingRequest{})
	if err != nil {
		gw.logger.Errorf("Ping client gateway: %v", err)
		return err
	}
	gw.logger.Debugf("Ping client gateway done")
	return nil
}

File: python/lib/control/gateway.py
from ..domain.contract import Contract
from ..api.proto import coreservice_pb2, coreservice_pb2_grpc

class Gateway(Contract):
    def __init__(self, channel):
        self.core_stub = coreservice_pb2_grpc.CoreServiceStub(channel)

    def Ping(self) -> str:
        self.core_stub.Ping(coreservice_pb2.PingRequest())
